### ISSUE TITLE
In autoupgrade process, Doctrine schema update process is performed after scripts which create inconsistent behavior

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#22294]
#### Describe the bug

Module autoupgrade [CoreUpgrader17:: upgradeDb()](https://github.com/PrestaShop/autoupgrade/blob/v4.11.0/classes/UpgradeTools/CoreUpgrader/CoreUpgrader17.php#L53)
```
    protected function upgradeDb($oldversion)
    {
        parent::upgradeDb($oldversion);

        $commandResult = $this->container->getSymfonyAdapter()->runSchemaUpgradeCommand();
        if (0 !== $commandResult['exitCode']) {
            throw (new UpgradeException($this->container->getTranslator()->trans('Error upgrading Doctrine schema', array(), 'Modules.Autoupgrade.Admin')))
                ->setQuickInfos(explode("\n", $commandResult['output']));
        }
    }
```

`runSchemaUpgradeCommand()` relies on Doctrine `prestashop:schema:update` (which is a very bad idea).

## An issue about which is first

If we wait for `runSchemaUpgradeCommand()` to create some columns, __these columns cannot be modified by scripts run in__ `parent::upgradeDb($oldversion);`! **Any column that is being managed by Doctrine cannot be modified in our scripts!**

## It fails for < 1.7.5 to 1.7.7

When I run autoupgrade CLI from PS 1.7.5 to PS 1.7.7 I have this
```
WARNING - 
            <div class="upgradeDbError">
            [WARNING] SQL 1.7.7.0
            1054 in UPDATE `ps_admin_filter` SET `filter_id` = SUBSTRING(`filter_id`, 1, 191): Unknown column 'filter_id' in 'field list'</div>

WARNING - 
            <div class="upgradeDbError">
            [WARNING] SQL 1.7.7.0
            1054 in ALTER TABLE `ps_admin_filter` CHANGE `filter_id` `filter_id` VARCHAR(191) NOT NULL: Unknown column 'filter_id' in 'ps_admin_filter'</div>
```

This is because column `filter_id` is created by Doctrine so it cannot be modified by the SQL scripts before.

#### Steps to Reproduce

1. Install PS 1.7.5
2. Run CLI autoupgrade
3. See warnings like https://travis-ci.com/github/matks/autoupgrade/jobs/451118636

**Comment by hibatallahAouadni** [comment]
Hello @matks 

Thanks for your report :smile: 
Could you please specify the severity of this bug :sweat_smile: 

Thanks in advance!

**Comment by atomiix** [comment]
Fixed by #22274
